langchain/document_loaders/blob_transformers/youtube_to_audio.py

```diff
@@ -0,0 +1,33 @@
+from langchain.document_loaders.base import BaseBlobTransformer
+from langchain.document_loaders.blob_loaders import Blob
+
+
+class YoutubeToAudioTransformer(BaseBlobTransformer):
+
+    """Dump YouTube url as audio file."""
+
+    # TODO: Output should be the blob path?
+    def lazy_transform(self, input_url: Blob) -> Blob:
+        """Lazily transform the blob."""
+
+        import yt_dlp
```

Comment:
We can move this code into a blob generator, which accepts a list of urls and yields video blobs.

Reply:
minor nit on this: `yt_dlp` downloads from url to audio directly w/o the video file intermediate.

Reply:
Ah I see!